wagtail/admin/templates/wagtailadmin/shared/button.html

```diff
@@ -1,7 +1,17 @@
 {% load wagtailadmin_tags %}
-<a href="{{ button.url }}" class="{{ button.classname }}" {{ button.base_attrs_string }}>
-    {% if button.icon_name %}
-        {% icon name=button.icon_name %}
-    {% endif %}
-    {{ button.label }}
-</a>
+
+{% if button.url %}
+    <a href="{{ button.url }}" class="{{ button.classname }}" {{ button.base_attrs_string }}>
+        {% if button.icon_name %}
+            {% icon name=button.icon_name %}
+        {% endif %}
+        {{ button.label }}
+    </a>
+{% else %}
+    <button type="button" class="{{ button.classname }}" {{ button.base_attrs_string }}>
```

Comment:
If the `type` is `button`, I believe this will NOT trigger a POST request and will just do nothing. This may be what we want sometimes but we probably need to think a bit better about how this is provided.

https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Elements/button#type

Maybe we do a check to see if `type` has been set and if not, set it to `button` in `base_attrs_string`?